### ISSUE TITLE
Sorted by holder count filters "tokens of 0 holders" out

### ISSUE THREAD

**Issue by Keith-CY** [Magickbase/godwoken_explorer#835]
![image](https://user-images.githubusercontent.com/7271329/183050046-7fc9d5ac-e654-4476-b051-13dd60de24d2.png)
![image](https://user-images.githubusercontent.com/7271329/183050105-906f4c76-9472-4c9c-ad83-17a108dbb1e9.png)

**Comment by FrederLu** [comment]
<img width="1118" alt="image" src="https://user-images.githubusercontent.com/25676576/188106810-b420dfde-9cff-42ea-9c95-b70f0c4b879c.png">
<img width="1092" alt="image" src="https://user-images.githubusercontent.com/25676576/188106902-bdb8040a-5a97-4c25-bbfe-f52baa661c01.png">
1、Data that needs to be kept as 0 when filtering by sort_value: EX_HOLDERS_COUNT.

<img width="1103" alt="image" src="https://user-images.githubusercontent.com/25676576/188109814-a72cfd28-7d57-4280-89e4-f5b960ca3215.png">
2、When sorting according to sort: [{ sorttype: ASC, sort_value: EX HOLDERS COUNT }], the data is not output as required.

**Comment by FrederLu** [comment]
![Image](https://user-images.githubusercontent.com/25676576/190071154-d7e4a492-b29c-4059-8513-885158b74731.png)
![Image](https://user-images.githubusercontent.com/25676576/190071059-b1ba0fb7-fdc7-4f9a-bf1f-8c5241285bc7.png)
1、After sorting the data using the ID and EX_HOLDERS_COUNT conditions in the ERC721_UDT interface, it is found that the data with holders 0 is not filtered, and the data is normal.


![Image](https://user-images.githubusercontent.com/25676576/190073377-443d3dea-cd71-4377-986a-3afb1f0dd4c7.png)
![Image](https://user-images.githubusercontent.com/25676576/190074090-cd61ca18-6c40-4ccb-9661-548495d1faa6.png)
![Image](https://user-images.githubusercontent.com/25676576/190074199-1de88813-bb33-4ba2-8e1c-2415b1b15df5.png)

Using `ID` sorting in the `ERC721_UDT` interface found that there is a large value in holders_count, but there is a problem with comparing the data through `EX_HOLDERS_COUNT` sorting (ASC\DESC). The problem is that `holders_count` starts at 1 instead of 0 according to ASC, and 0 again according to DESC.
If the `ERC721_UDT` interface has holders_count sorting requirements, it needs to be repaired immediately, otherwise the repair can be postponed.

**Comment by Naupio** [comment]
fixed in stg env @FrederLu 
erc20 example
```
    query {
      udts(
        input: {
          limit: 10
          sorter: [{ sort_type: DESC, sort_value: EX_HOLDERS_COUNT }]	
        }
      ) {
        entries {
          id
          name
          type
          supply
          holders_count
          account {
            eth_address
            script_hash
          }
        }
        metadata {
          total_count
          after
          before
        }
      }
    }

{
  "data": {
    "udts": {
      "entries": [
        {
          "account": {
            "eth_address": "0xe05d380839f32bc12fb690aa6fe26b00bd982613",
            "script_hash": "0xb66619b90d046b27c4724ab01808020ca0cbd0b5f801a5f7e9822ada4da797ab"
          },
          "holders_count": 27667,
          "id": 375,
          "name": "pCKB",
          "supply": "51601816771236920849617690",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": null,
            "script_hash": "0x595cc14e574a708dc70a320d2026f79374246ed4659261131cdda7dd5814b5ca"
          },
          "holders_count": 27667,
          "id": 1,
          "name": "pCKB (via Godwoken Bridge from CKB)",
          "supply": "51601816771236920849617690",
          "type": "BRIDGE"
        },
        {
          "account": {
            "eth_address": "0x975ab64f4901af5f0c96636dea0b9de3419d0c2f",
            "script_hash": "0xb01d03bbed4d9b55cfc484a3329875df13832c64e53c554233e18231026da891"
          },
          "holders_count": 10001,
          "id": 63191,
          "name": "CKB",
          "supply": "51601816771236920849617690",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": "0xdd9dbd70b18f99389f0fa51a8d416cc8912da598",
            "script_hash": "0x4fd88a761a28c059776acbf1d84cb0b76948c4fe8d419c0bc3bf2b03ec250cbb"
          },
          "holders_count": 3587,
          "id": 16025,
          "name": "pCKB",
          "supply": "51601816771236920849617690",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": "0x542ff77958f5b8ce319a5929d36a3ab78fa5fc78",
            "script_hash": "0x24f82151eebc0e841d449b81c3c0c7e241bfca705e630b755c496d2986ac081c"
          },
          "holders_count": 3125,
          "id": 16482,
          "name": "pCKB",
          "supply": "51601816771236920849617690",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": "0x73b52dcecebdef1e6d0d923742c5ad1fceaece25",
            "script_hash": "0xfcc062518ffa317b04bc0117fe633af3f0e45edfe0247b9a03725ebd89c1a222"
          },
          "holders_count": 802,
          "id": 16896,
          "name": "CKB",
          "supply": "51601816771236920849617690",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": "0x20fb98bb94ad9b98c0f0089138012e49323d0fef",
            "script_hash": "0xf4d317748f716b4a7c25b4b31f79ad424ef5c4e481c967a9ac7d70bbf6a3b1d7"
          },
          "holders_count": 392,
          "id": 40446,
          "name": "Wrapped USDC (via Force Bridge from Ethereum)",
          "supply": "145006099670",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": null,
            "script_hash": "0x454843ef13a5c57f97270b50bc49ef2d12e35fb7e72a145b0909ba65a5f9bfdd"
          },
          "holders_count": 392,
          "id": 29407,
          "name": "USDC (via ForceBridge from ETH)",
          "supply": "145006099670",
          "type": "BRIDGE"
        },
        {
          "account": {
            "eth_address": "0x630acc0a29e325ce022563df69ba7e25eeb1e184",
            "script_hash": "0xc9a0d07db60c1e5a07c1cb597aecb98aec8064f49f86336a76f80076386b382b"
          },
          "holders_count": 389,
          "id": 29450,
          "name": "ckUSDC",
          "supply": "145006099670",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": null,
            "script_hash": "0xf864e1aecdec0ca2ebc8762034e1b4b318ad8a49a73db20069f873ddf6759312"
          },
          "holders_count": 381,
          "id": 29378,
          "name": "DAI (via ForceBridge from ETH)",
          "supply": "164075189971000000000000",
          "type": "BRIDGE"
        }
      ],
      "metadata": {
        "after": "g3QAAAABaAJkAAl1X2hvbGRlcnNkAA1ob2xkZXJzX2NvdW50YgAAAX0=",
        "before": null,
        "total_count": 3893
      }
    }
  }
}
```

**Comment by Naupio** [comment]
erc721 example:
```
query {
  erc721_udts(
    input: {
      limit: 10
      sorter: [{ sort_type: DESC, sort_value: EX_HOLDERS_COUNT }]
    }
  ) {
    entries {
      id
      name
      contract_address_hash
      eth_type
      holders_count
      minted_count
    }
    metadata {
      total_count
      after
      before
    }
  }
}

{
  "data": {
    "erc721_udts": {
      "entries": [
        {
          "contract_address_hash": "0x76ef87aaaa9213b1e7f96460e23bd9d0b7dbcc64",
          "eth_type": "ERC721",
          "holders_count": 1466,
          "id": 70030,
          "minted_count": 2000,
          "name": "NervosPunkz"
        },
        {
          "contract_address_hash": "0x784cd3c52813098763c371df8fbe8ed27d2c1ebd",
          "eth_type": "ERC721",
          "holders_count": 308,
          "id": 58460,
          "minted_count": 2000,
          "name": "NervosPunkz"
        },
        {
          "contract_address_hash": "0x840909e2149c1a359b4c1129abe10dcd1beab979",
          "eth_type": "ERC721",
          "holders_count": 13,
          "id": 47308,
          "minted_count": 30,
          "name": "COLONIST"
        },
        {
          "contract_address_hash": "0x8c877b0b12d858043adcf4a7449f4c40775ef6fa",
          "eth_type": "ERC721",
          "holders_count": 11,
          "id": 41305,
          "minted_count": 11,
          "name": "COLONIST"
        },
        {
          "contract_address_hash": "0xc4a0c9f80c8a99a13e786f5141f3ea78fd1762f3",
          "eth_type": "ERC721",
          "holders_count": 11,
          "id": 24207,
          "minted_count": 13,
          "name": "SDV25-Deed"
        },
        {
          "contract_address_hash": "0x254960d5c6ab0befaf32e5f70896e1f02caaed0b",
          "eth_type": "ERC721",
          "holders_count": 9,
          "id": 46926,
          "minted_count": 20,
          "name": "COLONIST"
        },
        {
          "contract_address_hash": "0xe298acf9ec24283d359505182ec6f323b6b68e37",
          "eth_type": "ERC721",
          "holders_count": 8,
          "id": 24275,
          "minted_count": 11,
          "name": "MetaForest"
        },
        {
          "contract_address_hash": "0x9414d4ecc96dc464cd55665387420bcdb24d58da",
          "eth_type": "ERC721",
          "holders_count": 5,
          "id": 54610,
          "minted_count": 269,
          "name": "PsyMons"
        },
        {
          "contract_address_hash": "0x70ee76691bdd9696552af8d4fd634b3cf79dd529",
          "eth_type": "ERC721",
          "holders_count": 5,
          "id": 59098,
          "minted_count": 25,
          "name": "MNFT-0x53b9a6b381e5f02408ab81bea5462c179f475b7b-12"
        },
        {
          "contract_address_hash": "0x11a81159660a09da8de21ba1bd7ef78015279aad",
          "eth_type": "ERC721",
          "holders_count": 4,
          "id": 43037,
          "minted_count": 3,
          "name": "COL"
        }
      ],
      "metadata": {
        "after": "g3QAAAABaAJkAAl1X2hvbGRlcnNkAA1ob2xkZXJzX2NvdW50YQQ=",
        "before": null,
        "total_count": 3204
      }
    }
  }
}
```

**Comment by FrederLu** [comment]
https://github.com/Magickbase/godwoken_explorer/issues/835#issuecomment-1246584730 ERC20_udt sorted by holder count filters "tokens of 0 holders" out now works fine.
<img width="1090" alt="image" src="https://user-images.githubusercontent.com/25676576/190309198-36f46853-311f-40ef-a48a-f6b0fc6ffd22.png">
<img width="1128" alt="image" src="https://user-images.githubusercontent.com/25676576/190309250-d510add0-9e51-4c4e-b81d-b9e42007da14.png">



---
@Naupio 
When sorting according to sort: [{ sorttype: ASC, sort_value: EX HOLDERS COUNT }], the data is not output as required and still exists.
<img width="1123" alt="image" src="https://user-images.githubusercontent.com/25676576/190309770-2b1a3474-7fc2-4573-8c92-f5fc4786a724.png">
<img width="1169" alt="image" src="https://user-images.githubusercontent.com/25676576/190308695-48970a52-4fb2-4b1d-9a45-2c8ed0d127df.png">
<img width="1177" alt="image" src="https://user-images.githubusercontent.com/25676576/190310473-b5b455bb-960c-4ec0-8d4e-3c3c3324d6da.png">

**Comment by Naupio** [comment]
fixed 
```
query {
  udts(
    input: {
      limit: 10
      sorter: [{ sort_type: ASC, sort_value: EX_HOLDERS_COUNT }]
      after: "g3QAAAACZAACaWRiAAAYqWgCZAAJdV9ob2xkZXJzZAANaG9sZGVyc19jb3VudGEA"
    }
  ) {
    entries {
      id
      name
      type
      supply
      holders_count
      account {
        eth_address
        script_hash
      }
    }
    metadata {
      total_count
      after
      before
    }
  }
}

{
  "data": {
    "udts": {
      "entries": [
        {
          "account": {
            "eth_address": "0xd72223a3ea11a04d5ba84222f2cbec213729433e",
            "script_hash": "0xf357286ffe3ea6ab47a6e0aaba5c89a0ab9dfb1ec5e21145d34f51c4a860709d"
          },
          "holders_count": 0,
          "id": 6365,
          "name": "Name",
          "supply": "0",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": "0x191604813dffa08826ebc62bbfb961d53e046e7d",
            "script_hash": "0x69e2587bda0a962d5b263736fba67dd3c241370ed8a7598822221c195634b9fa"
          },
          "holders_count": 0,
          "id": 6366,
          "name": "Name",
          "supply": "0",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": "0xc5a48dda517f3116e8f48e3f6ae74e37c35d4aa5",
            "script_hash": "0x2e6e3b09fc5b614686820f64ec11ac928bb7a4a2cd6ae5197d8ba999821cfc82"
          },
          "holders_count": 0,
          "id": 6368,
          "name": "Name",
          "supply": "0",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": "0x46c573e6f4d83e75e6fea105345e4401703f4391",
            "script_hash": "0x0190a1333859335f51743d6f1016fbd740720d9d8ecd67cd48c76e3642479179"
          },
          "holders_count": 0,
          "id": 6369,
          "name": "Name",
          "supply": "0",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": "0xe7a15bbffd5f4b9ab3d412c25316ea94899f6140",
            "script_hash": "0x949ce83de572a0821660cd6e9126f21377da26c73d04a31cfc70b3346778296f"
          },
          "holders_count": 0,
          "id": 6485,
          "name": "Name",
          "supply": "0",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": "0x99e0e4932f3decb0bc6aa089e4fb2896050d83cb",
            "script_hash": "0x730469fa80500177edb3a4b23560fb5f8f3ffb65242a8134b697b9d9bea6897f"
          },
          "holders_count": 0,
          "id": 6486,
          "name": "Name",
          "supply": "0",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": "0x82d44fa20453d7bfc39c787b49d0be2ea963ff8e",
            "script_hash": "0xdda1d2b16467b73e08b2b47db337d73b26ea953d4d1c0e3b5dfc31921345b2ad"
          },
          "holders_count": 0,
          "id": 6487,
          "name": "Name",
          "supply": "0",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": "0x8cfaa4f7e9d0328b676548a98fe55f2f46baf901",
            "script_hash": "0x20639b142cfbb4e8bc9ae71a24151ceafd6e0b99be6843f9a8d34962fb541e52"
          },
          "holders_count": 0,
          "id": 6488,
          "name": "Name",
          "supply": "0",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": "0x1ccc4214d5a671772abce0bfdf680d2924b56e35",
            "script_hash": "0xbc6119944d72c5ccf15bb993f4838e6c6783d05eebeea020c5354149b838a64d"
          },
          "holders_count": 0,
          "id": 6673,
          "name": "Name",
          "supply": "0",
          "type": "NATIVE"
        },
        {
          "account": {
            "eth_address": "0x7259d520c2e24143caa982fc1bc60ef5cbe4dc4f",
            "script_hash": "0x2ac077f957df54053400d0d1cc5c84dd90d85e1d2c38b080e4a43787242c07e7"
          },
          "holders_count": 0,
          "id": 19195,
          "name": "Jio BUSD/DAI/USDC/USDT",
          "supply": "0",
          "type": "NATIVE"
        }
      ],
      "metadata": {
        "after": "g3QAAAACZAACaWRiAABK-2gCZAAJdV9ob2xkZXJzZAANaG9sZGVyc19jb3VudGEA",
        "before": "g3QAAAACZAACaWRiAAAY3WgCZAAJdV9ob2xkZXJzZAANaG9sZGVyc19jb3VudGEA",
        "total_count": 3917
      }
    }
  }
}
```

**Comment by FrederLu** [comment]
Verified.
